aws/src/integration/java/org/apache/iceberg/aws/glue/TestGlueCatalogTable.java

```diff
@@ -221,6 +221,69 @@ public void testUpdateTable() {
     assertThat(response.table().description()).isEqualTo(updatedComment);
   }
 
+  @Test
+  public void testDropColumn() {
+    String namespace = createNamespace();
+    String tableName = createTable(namespace);
+    Table table = glueCatalog.loadTable(TableIdentifier.of(namespace, tableName));
+    table
+        .updateSchema()
+        .addColumn("c2", Types.StringType.get(), "updated from Iceberg API")
+        .addColumn("c3", Types.StringType.get())
+        .commit();
+
+    updateTableColumns(
+        namespace,
+        tableName,
+        column -> {
+          if (column.name().equals("c3")) {
+            return column.toBuilder().comment("updated from Glue API").build();
+          } else {
+            return column;
+          }
+        });
+
+    table.updateSchema().deleteColumn("c2").deleteColumn("c3").commit();
```

Comment:
@lawofcycles For my own understanding but when you do the deleteColums and commit shouldn't this sync to catalog and these columns not be present anymore from glue? Or is this commit only changing the iceberg metadata in S3, and thus thats why the assertion is working correctly below as the catalog still has not the updated view of the table and contains the comments?

Reply:
@rahil-c  
> Or is this commit only changing the iceberg metadata in S3

Yes.
On current Iceberg's Glue integration, Glue Data Catalog holds all the column info that are registered in Iceberg Table even if these are deleted. IcebergToGlueConverter always try to sync all the historical column to Glue Data Catalog every time schema is updated.
For example, following scenario is expected behaviour.
 
1. Create Table with column x, y, z with Iceberg DDL using Glue Data Catalog as Iceberg Catalog
2. drop column y with Iceberg DDL  -> column y remain on Glue Data Catalog
3. drop column y, z from Glue Data Catalog API
4. add column y2 with Iceberg DDL  ->  column y, z appeared again on Glue Data Catalog in addition to column y2

I think those Information will be used to refer the historical columns on AWS Console when user query tables from such as Athena.

Reply:
Thanks for the explanation!